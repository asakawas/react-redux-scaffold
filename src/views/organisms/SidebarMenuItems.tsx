import React, { FunctionComponent } from 'react';
import List from '@material-ui/core/List';
import MusicNoteIcon from '@material-ui/icons/MusicNote';
import PeopleIcon from '@material-ui/icons/People';
import StarsIcon from '@material-ui/icons/Stars';
import SidebarMenuItem, { SidebarMenuItemProps } from '~/views/molecules/SidebarMenuItem';

/**
 * The type of props of SidebarMenuItems.
 */
export type SidebarMenuItemsProps = Readonly<{
  /** The event handler called when the Play Zundoko link is clicked */
  onClickPlayZundoko: NonNullable<SidebarMenuItemProps['onClick']>;

  /** The event handler called when the Users link is clicked */
  onClickUsers: NonNullable<SidebarMenuItemProps['onClick']>;

  /** The event handler called when the Kiyoshies link is clicked */
  onClickKiyoshies: NonNullable<SidebarMenuItemProps['onClick']>;
}>;

const SidebarMenuItems: FunctionComponent<SidebarMenuItemsProps> = ({
  onClickPlayZundoko,
  onClickUsers,
  onClickKiyoshies,
}) => (
  <>
    <List>
      <SidebarMenuItem text="Play Zundoko" icon={<MusicNoteIcon />} onClick={onClickPlayZundoko} />
    </List>
    <List>
      <SidebarMenuItem text="Users" icon={<PeopleIcon />} onClick={onClickUsers} />
    </List>
    <List>
      <SidebarMenuItem text="Kiyoshies" icon={<StarsIcon />} onClick={onClickKiyoshies} />
    </List>
  </>
);

export default React.memo(SidebarMenuItems);
