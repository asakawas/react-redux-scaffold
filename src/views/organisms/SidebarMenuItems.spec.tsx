import React from 'react';
import { render } from '@testing-library/react';
import SidebarMenuItems from './SidebarMenuItems';

describe('SidebarMenuItems', () => {
  test('is rendered as expected', () => {
    const { asFragment } = render(<SidebarMenuItems />);
    expect(asFragment()).toMatchSnapshot();
  });
});
