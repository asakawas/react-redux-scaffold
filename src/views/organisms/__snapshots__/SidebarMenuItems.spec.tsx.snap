// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarMenuItems is rendered as expected 1`] = `
<DocumentFragment>
  <ul
    class="MuiList-root MuiList-padding"
  >
    <div
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiListItemIcon-root"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 3v10.55c-.59-.34-1.27-.55-2-.55-2.21 0-4 1.79-4 4s1.79 4 4 4 4-1.79 4-4V7h4V3h-6z"
          />
        </svg>
      </div>
      <div
        class="MuiListItemText-root MuiListItemText-dense"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-h6"
        >
          Play Zundoko
        </span>
      </div>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
  </ul>
  <ul
    class="MuiList-root MuiList-padding"
  >
    <div
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiListItemIcon-root"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M16 11c1.66 0 2.99-1.34 2.99-3S17.66 5 16 5c-1.66 0-3 1.34-3 3s1.34 3 3 3zm-8 0c1.66 0 2.99-1.34 2.99-3S9.66 5 8 5C6.34 5 5 6.34 5 8s1.34 3 3 3zm0 2c-2.33 0-7 1.17-7 3.5V19h14v-2.5c0-2.33-4.67-3.5-7-3.5zm8 0c-.29 0-.62.02-.97.05 1.16.84 1.97 1.97 1.97 3.45V19h6v-2.5c0-2.33-4.67-3.5-7-3.5z"
          />
        </svg>
      </div>
      <div
        class="MuiListItemText-root MuiListItemText-dense"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-h6"
        >
          Users
        </span>
      </div>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
  </ul>
  <ul
    class="MuiList-root MuiList-padding"
  >
    <div
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiListItemIcon-root"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zm4.24 16L12 15.45 7.77 18l1.12-4.81-3.73-3.23 4.92-.42L12 5l1.92 4.53 4.92.42-3.73 3.23L16.23 18z"
          />
        </svg>
      </div>
      <div
        class="MuiListItemText-root MuiListItemText-dense"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-h6"
        >
          Kiyoshies
        </span>
      </div>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
  </ul>
</DocumentFragment>
`;
